scripts/background.js

```diff
@@ -335,116 +335,114 @@ chrome.runtime.onMessage.addListener(function (message, sender, sendResponse) {
+            url = tabs[0].url;
+            tabId = tabs[0].id;
+            if (url.includes('www.amazon')) {
+              fetch('https://archive.org/services/context/amazonbooks?url=' + url)
+                .then(resp => resp.json())
+                .then(resp => {
+                  if (('metadata' in resp && 'identifier' in resp['metadata']) ||
+                    'ocaid' in resp) {
+                    chrome.browserAction.setBadgeText({ tabId: tabId, text: 'B' })
                   }
-                });
-              }
-            } else if (event1.show_context === "singlewindow") {
+                })
+            }
+          })
+        }
+        if (event1.auto_update_context === true) {
+          if (event1.show_context === "tab") {
+            if (contexts.findIndex(e => e.tab !== 0) >= 0 || windowIdtest !== 0) {
               chrome.tabs.query({
-                windowId: windowIdSingle
+                windowId: windowIdtest
               }, function (tabs) {
                 var tab1 = tabs[0];
-                if (tabId !== tab1.id && tab.id !== tab1.id) {
-                  chrome.tabs.update(tab1.id, { url: chrome.runtime.getURL("singleWindow.html") + "?url=" + singlewindowurl });
+                tabIdtest = tab1.id;
+                if (tab.id !== tabIdtest && contexts.findIndex(e => e.tab !== tab.id) >= 0 && tabs.filter(e => e.id === tabId).length === 0 && contexts.filter(e => e.tab === tabId).length === 0 && contexts.filter(e => e.tab === tab.id).length === 0) {
+                  for (var i = 0; i < contexts.length; i++) {
+                    var e = contexts[i];
+                    chrome.tabs.update(parseInt(e.tab), { url: e.htmlUrl + urlsToAppend[i] });
+                  };
                 }
               });
-            } else {
-              if (contexts.findIndex(e => e.window !== 0) >= 0) {
-                var i = 0;
-                contexts.map(e => {
-                  chrome.tabs.query({
-                    windowId: e.window
-                  }, function (tabs) {
-                    if (tabs.length > 0) {
-                      e.tabContextName = tabs[0].id;
-                      if (contexts.filter(e => e.tabContextName === tabId).length == 0 && contexts.filter(e => e.tabContextName === tab.id).length === 0) {
-                        chrome.tabs.update(e.tabContextName, { url: e.htmlUrl + urlsToAppend[i++] }, function (tab) { });
-                      }
+            }
+          } else if (event1.show_context === "singlewindow") {
+            chrome.tabs.query({
+              windowId: windowIdSingle
+            }, function (tabs) {
+              var tab1 = tabs[0];
+              if (tabId !== tab1.id && tab.id !== tab1.id) {
+                chrome.tabs.update(tab1.id, { url: chrome.runtime.getURL("singleWindow.html") + "?url=" + singlewindowurl });
+              }
+            });
+          } else {
+            if (contexts.findIndex(e => e.window !== 0) >= 0) {
```

Comment:
Inside `} else {`, there is only this `if` statement and nothing else.
Can you make it `} else if (contexts.findIndex(e => e.window !== 0) >= 0) {` to simplify even more?